src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/Streaming/FundingSource/FundingSource.tsx

```diff
@@ -0,0 +1,283 @@
+import React, { ReactNode, useCallback } from 'react';
+import { defineMessages } from 'react-intl';
+import classNames from 'classnames';
+import { parseInt } from 'lodash';
+import { ROOT_DOMAIN_ID } from '@colony/colony-js';
+
+import { Colony } from '~data/index';
+import {
+  FormSection,
+  Input,
+  InputLabel,
+  SelectHorizontal,
+  SelectOption,
+  TokenSymbolSelector,
+} from '~core/Fields';
+import DomainDropdown from '~core/DomainDropdown';
+import { COLONY_TOTAL_BALANCE_DOMAIN_ID } from '~constants';
+import ColorTag, { Color } from '~core/ColorTag';
+import TokenIcon from '~dashboard/HookedTokenIcon';
+
+import { FundingSource as FundingSourceType } from '../types';
+
+import styles from './FundingSource.css';
+
+const MSG = defineMessages({
+  team: {
+    id: 'dashboard.ExpenditurePage.Streaming.FundingSource.team',
+    defaultMessage: 'Team',
+  },
+  rate: {
+    id: 'dashboard.ExpenditurePage.Streaming.FundingSource.rate',
+    defaultMessage: 'Rate',
+  },
+  notSet: {
+    id: 'dashboard.ExpenditurePage.Streaming.FundingSource.notSet',
+    defaultMessage: 'Not set',
+  },
+  time: {
+    id: 'dashboard.ExpenditurePage.Streaming.FundingSource.time',
+    defaultMessage: 'Rate time',
+  },
+  month: {
+    id: 'dashboard.ExpenditurePage.Streaming.FundingSource.month',
+    defaultMessage: 'month',
+  },
+  week: {
+    id: 'dashboard.ExpenditurePage.Streaming.FundingSource.week',
+    defaultMessage: 'week',
+  },
+  day: {
+    id: 'dashboard.ExpenditurePage.Streaming.FundingSource.day',
+    defaultMessage: 'day',
+  },
+  hour: {
+    id: 'dashboard.ExpenditurePage.Streaming.FundingSource.hour',
+    defaultMessage: 'hour',
+  },
+  limit: {
+    id: 'dashboard.ExpenditurePage.Streaming.FundingSource.limit',
+    defaultMessage: 'Limit',
+  },
+});
+
+const displayName = 'dashboard.ExpenditurePage.Streaming.FundingSource';
+
+enum timePeriod {
```

Comment:
The convention in our code base is to have enum starting with capital letter. Please, update.